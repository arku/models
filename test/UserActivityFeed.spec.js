const mongoose = require('mongoose');
const {
  User,
  UserActivityFeedEvents: {
    UserActivityFeedEvent,
    AcademicProfileCommentEvent,
    ReviewAnswerEvent,
  },
} = require('../')(mongoose);

describe('UserActivityFeedEvent', () => {
  beforeAll(async () => {
    await mongoose.connect(global.__MONGO_URI__, { useNewUrlParser: true });
  });

  afterAll(async () => {
    await mongoose.disconnect();
  });

  beforeEach(async () => {
    await mongoose.connection.db.dropDatabase();
  });

  it('should fail with missing props', () => {
    const event = new UserActivityFeedEvent();
    return event.save()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should save successfully with appropriate fields', () => {
    const event = new UserActivityFeedEvent({
      doc: mongoose.Types.ObjectId(),
    });

    return event.save()
      .then((result) => {
        expect(typeof result.createdAt).toBe('object');
        expect(result.type).toBe('UserActivityFeedEvent');
      });
  });
});

describe('AcademicProfileCommentEvent', () => {
  let user;

  beforeAll(async () => {
    await mongoose.connect(global.__MONGO_URI__, { useNewUrlParser: true });
  });

  afterAll(async () => {
    await mongoose.disconnect();
  });

  beforeEach(async () => {
    await mongoose.connection.db.dropDatabase();
    await User.createIndexes();
    user = new User({
      uid: 'abcdefghijklmnopqrstuvwxyz',
      email: 'someone@somewhere.com',
      name: 'Someone',
    });

    await user.save();
  });

  it('should fail with missing props', () => {
    const event = new AcademicProfileCommentEvent();
    return event.save()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should save successfully with appropriate fields', () => {
    const event = new AcademicProfileCommentEvent({
      doc: mongoose.Types.ObjectId(),
      createdBy: user._id,
      commentType: 'tech',
      project: 'Cifrado Cesar',
      text: 'A test',
    });

    return event.save()
      .then((result) => {
        expect(typeof result.createdAt).toBe('object');
        expect(result.type).toBe('AcademicProfileCommentEvent');
      });
  });

  it('should fail when createdBy does not exist', async () => {
    const event = new AcademicProfileCommentEvent({
      doc: mongoose.Types.ObjectId(),
      createdBy: (new User())._id,
      commentType: 'tech',
      project: 'Cifrado Cesar',
      text: 'A test',
    });

    return event.save()
      .catch((err) => {
        expect(err.message).toBe('CreatedBy does not exist');
      });
  });
});

describe('ReviewAnswerEvent', () => {
  let user;

  beforeAll(async () => {
    await mongoose.connect(global.__MONGO_URI__, { useNewUrlParser: true });
  });

  afterAll(async () => {
    await mongoose.disconnect();
  });

  beforeEach(async () => {
    await mongoose.connection.db.dropDatabase();
    await User.createIndexes();
    user = new User({
      uid: 'abcdefghijklmnopqrstuvwxyz',
      email: 'someone@somewhere.com',
      name: 'Someone',
    });

    await user.save();
  });

  it('should fail with missing props', () => {
    const event = new ReviewAnswerEvent();
    return event.save()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should save successfully with appropriate fields', () => {
    const event = new ReviewAnswerEvent({
      doc: mongoose.Types.ObjectId(),
      createdBy: user._id,
      commentType: 'tech',
      project: 'Cifrado Cesar',
      text: 'A test',
    });

    return event.save()
      .then((result) => {
        expect(typeof result.createdAt).toBe('object');
        expect(result.type).toBe('ReviewAnswerEvent');
      });
  });

  it('should fail when createdBy does not exist', async () => {
    const event = new ReviewAnswerEvent({
      doc: mongoose.Types.ObjectId(),
      createdBy: (new User())._id,
      commentType: 'tech',
      project: 'Cifrado Cesar',
      text: 'A test',
    });

    return event.save()
      .catch((err) => {
        expect(err.message).toBe('CreatedBy does not exist');
      });
  });
});
