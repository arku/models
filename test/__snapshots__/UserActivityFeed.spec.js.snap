// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcademicProfileCommentEvent should fail with missing props 1`] = `
Object {
  "commentType": [ValidatorError: Path \`commentType\` is required.],
  "createdBy": [ValidatorError: Path \`createdBy\` is required.],
  "doc": [ValidatorError: Path \`doc\` is required.],
  "project": [ValidatorError: Path \`project\` is required.],
  "text": [ValidatorError: Path \`text\` is required.],
}
`;

exports[`ReviewAnswerEvent should fail with missing props 1`] = `
Object {
  "commentType": [ValidatorError: Path \`commentType\` is required.],
  "createdBy": [ValidatorError: Path \`createdBy\` is required.],
  "doc": [ValidatorError: Path \`doc\` is required.],
  "project": [ValidatorError: Path \`project\` is required.],
  "text": [ValidatorError: Path \`text\` is required.],
}
`;

exports[`UserActivityFeedEvent should fail with missing props 1`] = `
Object {
  "doc": [ValidatorError: Path \`doc\` is required.],
}
`;
